packages/stream-model-instance-handler/src/model-instance-document-handler.ts

```diff
@@ -312,4 +312,23 @@ export class ModelInstanceDocumentHandler implements StreamHandler<ModelInstance
       }
     }
   }
+
+  /**
+   *  Helper function to validate if locked fields are being mutated
+   */
+  async _validateLockedFieldsUpdate(model: Model, payload: any): Promise<void> {
+    if ('immutableFields' in model.content) {
+      // No locked fields
+      if (model.content.immutableFields.length == 0) return
+
+      for (const lockedField of model.content.immutableFields) {
+        const mutated = payload.data.some(
+          (entry) => entry.op === 'replace' && entry.path.split('/').pop() === lockedField
```

Comment:
hmm... this makes me nervous.  Are we sure that "replace" is the only way jsonPatch could update a field?  Like, what if the locked field is an array and we try to push a new value into that array?  Can we just error if the field shows up in the payload at all, regardless of "op" type?  Ideally I'd prefer if we didn't have to understand and parse the jsonPatch encoding format at all, but I don't know of a way to do a general-purpose type-aware equality check in javascript, so this might be the best we can do for now.

This also makes me realize we should also have some additional unit tests for different types of updates rather than just full replacement of a string. E.g: add a field to an array, add a field to a sub-object, remove the field, change the type of a value (like from the number `1` to the string "1", on a field where the schema from the Model allows the field to be a number or a string), etc.

Reply:
You are correct; we do not genuinely care about the type of mutation being performed. If it is present in the payload, we assume that an attempt to make a change is being made.